overlord/fdestate/fdemgr.go

```diff
@@ -132,19 +132,39 @@ func (m *FDEManager) ReloadModeenv() error {
 	return initModeFromModeenv(m)
 }
 
+type unlockedStateManager struct {
+	unlocker boot.Unlocker
+	m        *FDEManager
```

Comment:
this could be simplified to:

{
FDEManager
unlock  bool
}

as we just check if unlocker is nil